internal/cloud/cloudplan/saved_plan.go

```diff
@@ -16,13 +22,34 @@ type SavedPlanBookmark struct {
 
 func LoadSavedPlanBookmark(filepath string) (SavedPlanBookmark, error) {
 	bookmark := SavedPlanBookmark{}
-	data, err := os.ReadFile(filepath)
 
+	file, err := os.Open(filepath)
+	if err != nil {
+		fmt.Println("error opening file")
+		return bookmark, err
+	}
+	defer file.Close()
+
+	data, err := io.ReadAll(file)
 	if err != nil {
+		fmt.Println("error reading file")
 		return bookmark, err
 	}
 
-	err = json.Unmarshal([]byte(data), &bookmark)
+	e := json.Unmarshal([]byte(data), &bookmark)
+	if e != nil {
+		fmt.Println("could not unmarshal")
+		return bookmark, e
+	}
+
+	if bookmark.RemotePlanFormat != 1 {
+		return bookmark, ErrInvalidRemotePlanFormat
+	} else if bookmark.Hostname == "" {
+		return bookmark, ErrInvalidHostname
+	} else if bookmark.RunID == "" {
+		return bookmark, ErrInvalidRunID
```

Comment:
You could add some extra validations here:
- Ensure that hostname is a valid hostname
- Ensure that Run ID is in the correct format, you can use the following:
```go
func isResourceIDFormat(resourcePrefix string, id string) bool {
	base58Regex, err := regexp.Compile(fmt.Sprintf("^%s-[1-9A-HJ-NP-Za-km-z]{16}$", resourcePrefix))
	if err != nil {
		return false
	}
	return base58Regex.MatchString(id)
}
```
The resource prefix in this case would be `run`.

Reply:
This might be where @brandonc's feedback comes in handy. So I will write validation on the custom type and refactor.

Reply:
> You could add some extra validations here:
> 
> * Ensure that hostname is a valid hostname
> * Ensure that Run ID is in the correct format, you can use the following:
> 
> ```go
> func isResourceIDFormat(resourcePrefix string, id string) bool {
> 	base58Regex, err := regexp.Compile(fmt.Sprintf("^%s-[1-9A-HJ-NP-Za-km-z]{16}$", resourcePrefix))
> 	if err != nil {
> 		return false
> 	}
> 	return base58Regex.MatchString(id)
> }
> ```
> 
> The resource prefix in this case would be `run`.

How about using `strings.HasPrefix()`? Seems less verbose and does the same job. What do you think? @sebasslash 
How are we validating on hostname? What are we accepting (or not)? @nfagerlund